packages/@ember/-internals/runtime/lib/system/core_object.js

```diff
@@ -135,6 +135,49 @@ function initialize(obj, properties) {
 }
 
 /**
+  `CoreObject` is the base class for all Ember constructs. It establishes a
+  class system based on Ember's Mixin system, and provides the basis for the
+  Ember Object Model. `CoreObject` should generally not be used directly,
+  instead you should use `EmberObject`.
+
+  ## Usage
+
+  You can define a class by extending from `CoreObject` using the `extend`
+  method:
+
+  ```js
+  const Person = CoreObject.extend({
+    name: 'Tomster',
+  });
+  ```
+
+  For detailed usage, see the [Object Model](https://guides.emberjs.com/release/object-model/)
+  section of the guides.
+
+  ## Usage with Native Classes
+
+  Native Javascript `class` syntax can be used to extend from any `CoreObject`
+  based class:
+
+  ```js
+  class Person extends CoreObject {
+    init() {
+      super.init(...arguments);
```

Comment:
You may want to consider explicitly passing arguments to `super.init` in your example, so consumers know what `init()` receives.

Reply:
I don't think we actually guarantee a shape to the arguments to `init`, which is why the recommended advice was to always do `this._super(...arguments);`

Reply:
Ok, what if I make an ember object subclass, and call `new Foo({ name: 'Mike' })`. Is this exactly the same as `Ember.Object.create`, in that "stuff is just there" by the time the init hook is fired?

Reply:
`new Foo` is not supported, actually. We should probably call this out in the notes, but yes, if you do `Foo.create({ name: 'Mike' })` then the `name` property will be set by the time the `init` hook is fired.